assets/RNAseq_report.Rmd

```diff
@@ -341,19 +834,55 @@ The original plot can be downloaded [here](./differential_gene_expression/plots/
 
 \
 
-```{r, echo=F, out.width="80%", dpi=1200, fig.align='center'}
-knitr::include_graphics(paste0(wd,"/differential_gene_expression/plots/PCA_plot.svg"))
+```{r PCA_plots, echo=F, warning=F}
+############################ PCA PLOTS ########################
+pcaData <- plotPCA(if (params$rlog) rld else vsd,intgroup=c("combfactor"),ntop = dim(if (params$rlog) rld else vsd)[1], returnData=TRUE)
+percentVar <- round(100*attr(pcaData, "percentVar"))
+pca <- ggplot(pcaData, aes(PC1, PC2, color=combfactor)) +
+    geom_point(size=3) +
+    xlab(paste0("PC1: ",percentVar[1],"% variance")) +
+    ylab(paste0("PC2: ",percentVar[2], "% variance")) +
+    theme(legend.title = element_blank()) +
+    coord_fixed() +
+    theme_bw() +
+    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())
+suppressMessages(ggsave(plot = pca, filename = "differential_gene_expression/plots/PCA_plot.pdf", device = "pdf", dpi = 300))
+suppressMessages(ggsave(plot = pca, filename = "differential_gene_expression/plots/PCA_plot.svg", device = "svg", dpi = 150))
+suppressMessages(ggsave(plot = pca, filename = "differential_gene_expression/plots/PCA_plot.png", device = "png", dpi = 150))
+########################### PCA PLOT with batch-corrected data ############
+if(!isNot(params$batch_effect)){
```

Comment:
Can't this be done with an `eval` setting to at the beginning of the block?

Reply:
Then this if clause is not necessary, and this code can be moved to the next block (PCA_ifBatch_plot)